realm/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -86,10 +91,20 @@ public long getItemId(int i) {
      * Update the RealmResults associated to the Adapter. Useful when the query has been changed.
      * If the query does not change you might consider using the automaticUpdate feature
      *
-     * @param realmResults the new RealmResults coming from the new query.
+     * @param queryResults the new RealmResults coming from the new query.
      */
-    public void updateRealmResults(RealmResults<T> realmResults) {
-        this.realmResults = realmResults;
+    public void updateRealmResults(RealmResults<T> queryResults) {
+        if (listener != null) {
+            // Making sure that Adapter is refreshed correctly if new RealmResults come from another Realm
```

Comment:
I think you should write "another Realm instance" as "another Realm" make it sounds like it can come from another Realm file.